presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeRemoteFunctions.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.nativeworker;
+
+import com.facebook.presto.testing.QueryRunner;
+import com.facebook.presto.tests.AbstractTestQueryFramework;
+import org.testng.SkipException;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.nativeworker.NativeQueryRunnerUtils.createLineitem;
+
+public abstract class AbstractTestNativeRemoteFunctions
+        extends AbstractTestQueryFramework
+{
+    public String remoteFunctionServerUds;
+    public String remoteFunctionServerBinaryPath;
+
+    @Override
+    protected void preInitQueryRunners()
+    {
+        // Initialize the remote function thrift server process.
+        remoteFunctionServerBinaryPath = System.getProperty("REMOTE_FUNCTION_SERVER");
+        if (remoteFunctionServerBinaryPath != null) {
+            remoteFunctionServerUds = PrestoNativeQueryRunnerUtils.startRemoteFunctionServer(remoteFunctionServerBinaryPath);
+        }
+    }
+
+    @Override
+    protected void postInitQueryRunners()
+    {
+        // Install json function registration namespace manager.
+        PrestoNativeQueryRunnerUtils.setupJsonFunctionNamespaceManager(getQueryRunner(), PrestoNativeQueryRunnerUtils.REMOTE_FUNCTION_JSON_SIGNATURES, PrestoNativeQueryRunnerUtils.REMOTE_FUNCTION_CATALOG_NAME);
+    }
+
+    @Override
+    protected void createTables()
+    {
+        // Create some tpch tables.
+        QueryRunner queryRunner = (QueryRunner) getExpectedQueryRunner();
+        createLineitem(queryRunner);
+    }
+
+    @Test
+    public void testRemoteFunction()
+    {
+        if (remoteFunctionServerBinaryPath == null) {
+            throw new SkipException("Skipping remote functions tests since REMOTE_FUNCTION_SERVER was not provided.");
+        }
+
+        assertQuery("SELECT remote.schema.ceil(linenumber) FROM lineitem", "SELECT ceil(linenumber) FROM lineitem");
+        assertQuery("SELECT remote.schema.ceil(discount) FROM lineitem", "SELECT ceil(discount) FROM lineitem");
+        assertQuery("SELECT remote.schema.ceil(discount * 7) FROM lineitem", "SELECT ceil(discount * 7) FROM lineitem");
+
+        assertQuery("SELECT remote.schema.ceil(ceil(discount)) FROM lineitem", "SELECT ceil(ceil(discount * 7)) FROM lineitem");
+
+        // TODO: `throwOnError` needs to be implemented in the server so we can have remote functions as function parameters.
```

Comment:
What happens now? Maybe create a GitHub issue and include the current error in the description, then add a link to the issue here?

Reply:
Today we get this exception:

https://github.com/facebookincubator/velox/blob/main/velox/functions/remote/server/RemoteFunctionService.cpp#L79

I have this listed as one item in this Issue:

https://github.com/facebookincubator/velox/issues/5288